src/main/java/edu/umich/its/cpm/Utils.java

```diff
@@ -433,40 +433,93 @@ public static boolean isCurrentUserCPMAdmin(String userId, Environment env) {
 	}
 
 	/*
-	 * get CoSign user, for local development user will be passed from url parameter as
-	 * http://localhost:8080/?testUser=<uniqname> together with allow.testUser.urlOverride from the
-	 * application.properties we will determine valid user. In short this block is mimicking
-	 * cosign user if CoSign is not enabled
+	 * this block of code getting the logged in user to cpm. CPM features become user for admins, get User's when no
+	 * authentication is used using the flag "allow.testUser.urlOverride" along with testUser parameter turned on from URL.
+	 * All that logic is handled here.
+	 *
 	 */
 	public static String getUserLoginId(HttpServletRequest request, Environment env) {
 		String user = null;
 		String remoteUser = request.getRemoteUser();
 		String testUser = request.getParameter(TEST_USER);
-		boolean isTestUrlEnabled = Boolean.valueOf(env.getProperty(ALLOW_USER_URLOVERRIDE));
+		boolean allowTestUserOverride = ((env.getProperty(ALLOW_USER_URLOVERRIDE) == null ? false : Boolean.valueOf(env.getProperty(ALLOW_USER_URLOVERRIDE))));
 		String testUserInSession = (String) request.getSession().getAttribute(TEST_USER);
 
-		if (isTestUrlEnabled && testUser != null) {
-			user = testUser;
-			request.getSession().setAttribute(TEST_USER, testUser);
-			log.debug("TEST USER is " + user);
-			return user;
+		// simple authenticated user
+		if (!allowTestUserOverride) {
+			return remoteUser;
 		}
-		if (isTestUrlEnabled && testUserInSession != null) {
-			user = testUserInSession;
-			log.debug("TEST USER FROM SESSION " + user);
-			return user;
+
+		if (allowTestUserOverride) {
+
+			// user is authenticated, can become user using "testUser" attribute so we authorize the user using LDAP.
+			// only admin's can become users
+			if (remoteUser != null) {
```

Comment:
See what happens if you check just for isCurrentUserCPMAdmin at this level.  If it is false I believe you can just set user to makeUserRemote.... and return right away.  That would simplify the other nested ifs.